hipparchus-core/src/test/java/org/hipparchus/analysis/differentiation/DerivativeTest.java

```diff
@@ -0,0 +1,326 @@
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.cosh(value));
+        Assert.assertEquals(expectedOperation, actualOperation);
+    }
+
+    @Test
+    public void testSinh() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.sinh();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.sinh(value));
+        Assert.assertEquals(expectedOperation, actualOperation);
+    }
+
+    @Test
+    public void testAcos() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.acos();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.acos(value));
+        Assert.assertEquals(expectedOperation.getValue(), actualOperation.getValue(), TOLERANCE);
+    }
+
+    @Test
+    public void testFloor() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.floor();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.floor(value));
+        Assert.assertEquals(expectedOperation, actualOperation);
+    }
+
+    @Test
+    public void testCeil() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.ceil();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.ceil(value));
+        Assert.assertEquals(expectedOperation.getValue(), actualOperation.getValue(), TOLERANCE);
+    }
+
+    @Test
+    public void testRint() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.rint();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.rint(value));
+        Assert.assertEquals(expectedOperation, actualOperation);
+    }
+
+    @Test
+    public void testSign() {
+        // GIVEN
+        final double value = 0.5;
+        final TestDerivative testDerivative = new TestDerivative(value);
+        // WHEN
+        final TestDerivative actualOperation = testDerivative.sign();
+        // THEN
+        final TestDerivative expectedOperation = new TestDerivative(FastMath.signum(value));
+        Assert.assertEquals(expectedOperation.getValue(), actualOperation.getValue(), TOLERANCE);
+    }
+
+    static class TestDerivative implements Derivative<TestDerivative> {
+
```

Comment:
IS this for testing the default implementations (and ensuring test coverage)?

Reply:
Yes it's for coverage